Java/app/src/main/java/com/taboola/sdk3example/sdk_native/FeedFragment.java

```diff
@@ -0,0 +1,154 @@
+            mNativeUnit.fetchRecommendations(requestData, new TBLRecommendationRequestCallback() {
+                @Override
+                public void onRecommendationsFetched(TBLRecommendationsResponse tblRecommendationsResponse) {
+                    println("Taboola | fetchInitialContent | onRecommendationsFetched");
+                    // Send content to RecyclerView Adapter
+                    addRecommendationToFeed(tblRecommendationsResponse);
+                }
+
+                @Override
+                public void onRecommendationsFailed(Throwable throwable) {
+                    println(String.format("Taboola | onRecommendationsFailed: %s", throwable.getLocalizedMessage()));
+                }
+            });
+        }
+    }
+
+    /**
+     * Taboola "Native Integration" currently differentiates between first and subsequent content fetch calls.
+     * This method fetches additional items for this Feed implementation.
+     */
+    private void loadNextRecommendationsBatch() {
+        mNativeUnit.getNextRecommendationsBatch(null, new TBLRecommendationRequestCallback() {
+            @Override
+            public void onRecommendationsFetched(TBLRecommendationsResponse tblRecommendationsResponse) {
+                println("Taboola | loadNextRecommendationsBatch | onRecommendationsFetched");
+                addRecommendationToFeed(tblRecommendationsResponse);
+            }
+
+            @Override
+            public void onRecommendationsFailed(Throwable throwable) {
+                println(String.format("Taboola | onRecommendationsFailed: %s", throwable.getLocalizedMessage()));
+            }
+        });
+    }
+
+    /**
+     * This method parses the response with its items and adds them to the RecyclerView Adapter.
+     */
+    private void addRecommendationToFeed(TBLRecommendationsResponse recommendationsResponse) {
+        TBLPlacement placement = recommendationsResponse.getPlacementsMap().values().iterator().next();
+
+        if (placement != null) {
+            // Update data structure
+            mData.addAll(placement.getItems());
+
+            // Update data in RecyclerView adapter
+            int itemCount = mRecyclerView.getAdapter().getItemCount();
+            if (itemCount > 0) {
+                mRecyclerView.getAdapter().notifyItemRangeInserted(itemCount, placement.getItems().size());
+            }
+        }
+    }
+
+    /**
+     * Define a Page that represents this screen, get a Unit from it, add it to screen and fetch its content
+     * Notice: A Unit of unlimited items, called "Feed" in Taboola, can be set in TBL_PLACEMENT_TYPE.PAGE_BOTTOM only.
+     */
+    private void getTaboolaUnit() {
+        // Define a page to control all Unit placements on this screen
+        TBLNativePage nativePage = Taboola.getNativePage("text", "https://blog.taboola.com");
+
+        // Define a single Unit to display
+        mNativeUnit = nativePage.build("list_item", new TBLNativeListener() {
```

Comment:
I'm missing something here, the method signature for NativePage.build, in 3.x master, is:

`build(String placementName, @Nullable TBLPublisherInfo tblPublisherInfo, TBLNativeListener tblNativeListener, @Nullable TBLRequestData tblRequestData)`

Reply:
The same code as in Kotlin project it takes TBLPublisherInfo from global

Reply:
Done